File: src/com/pwootage/fasterflux/blocks/te/FFTileEntityBase.java
package com.pwootage.fasterflux.blocks.te;

import com.pwootage.fasterflux.FasterFlux;

import net.minecraft.world.World;
import Reika.DragonAPI.DragonAPICore;
import Reika.DragonAPI.Base.TileEntityBase;

public abstract class FFTileEntityBase extends TileEntityBase {

	public FFTileEntityBase() {
		super();
	}

	@Override
	protected void animateWithTick(World world, int x, int y, int z) {
	}

	@Override
	public int getRedstoneOverride() {
		return 0;
	}

	@Override
	public boolean shouldRenderInPass(int pass) {
		if (pass == 0) {
			return true;
		}
		return false;
	}
	
	@Override
	public int getPacketDelay() {
		if (DragonAPICore.isSinglePlayer()) {
			return 1; //No need to slow syncs here...
		}
		return Math.max(1, FasterFlux.conf.syncDelay);
	}
}

File: src/main/scala/com/pwootage/fasterflux/blocks/te/FFTileEntityBase.scala
package com.pwootage.fasterflux.blocks.te

import com.pwootage.fasterflux.FasterFlux
import net.minecraft.world.World
import Reika.DragonAPI.DragonAPICore
import Reika.DragonAPI.Base.TileEntityBase

abstract class FFTileEntityBase extends TileEntityBase() {

  protected override def animateWithTick(world: World, x: Int, y: Int, z: Int) {
  }

  override def getRedstoneOverride(): Int = 0

  override def shouldRenderInPass(pass: Int): Boolean = pass == 0

  override def getPacketDelay(): Int = DragonAPICore.isSinglePlayer match {
    case true => 1 //Don't need to throttle ever here, makes SP better
    case false => Math.max(1, FasterFlux.conf.syncDelay)
  } 
}